Core/TestsFoundation/DataSeeding/AuthenticationProviders/DataSeeder+DSAuthenticationProvider.swift

```diff
@@ -0,0 +1,26 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+extension DataSeeder {
+    @discardableResult
+    public func setSelfRegistrationForCanvas(selfRegistration: DSSelfRegistration) -> DSAuthenticationProvider {
+        let body = UpdateDSAuthenticationProviderRequest.Body(selfRegistration: selfRegistration)
+        let request = UpdateDSAuthenticationProviderRequest(body: body)
```

Comment:
nit: IMHO we could simply do it like this, without sacrificing readability:
```
let request = UpdateDSAuthenticationProviderRequest(body: .init(selfRegistration: selfRegistration))
```
(no need to change it unless you push something else anyway)

Reply:
I like your suggestion but wanted to stay consistent with other DataSeeding solutions so I'd leave it like this for now.